addons/payment/data/payment_method_data.xml

```diff
@@ -2288,6 +2288,28 @@
         />
     </record>
 
+    <record id="payment_method_paylater" model="payment.method">
```

Comment:
Same logic for Pay Later: the `name` can stay but the XMLID and `code` should be `paylater_india` if that specific PM is only available in India, which seems to be case based on its `supported_currency_ids|country_ids` fields.